docs/proposals/introduce-apps.md

```diff
@@ -2,125 +2,116 @@
   - [Goals](#goals)
   - [Non-Goals](#non-goals)
 - [Implementation](#implementation)
   - [Architecture](#architecture)
   - [Components](#components)
-    - [ApplicationDistributor](#applicationdistributor)
+    - [Application Definition Synchronizer](#application-definition-synchronizer)
     - [ApplicationDefinition CR](#applicationdefinition-cr)
-    - [Updates to KKP API](#updates-to-kkp-api)
     - [ApplicationInstallation CR](#applicationinstallation-cr)
-    - [ApplicationInstallationController](#applicationinstallationcontroller)
+    - [Application Installation Controller](#application-installation-controller)
+    - [Credentials management](#credentials-management)
+    - [Updates to KKP API](#updates-to-kkp-api)
   - [Common User Flows](#common-user-flows)
+    - [Cluster Administrator Installs a New Application](#cluster-administrator-installs-a-new-application)
+    - [Cluster Administrator Creates A new Cluster with Application](#cluster-administrator-creates-a-new-cluster-with-application)
+    - [Cluster Administrator Updates the version of an Application (supported only with kubectl)](#cluster-administrator-updates-the-version-of-an-application-supported-only-with-kubectl)
+    - [KKP Administrator Adds A New Version of an Application or a New Application](#kkp-administrator-adds-a-new-version-of-an-application-or-a-new-application)
+    - [KKP Administrator remove a Version of an Application](#kkp-administrator-remove-a-version-of-an-application)
   - [Alternatives considered](#alternatives-considered)
 - [Glossary](#glossary)
   - [Default Addon](#default-addon)
   - [Application](#application)
   - [Application by Kubermatic](#application-by-kubermatic)
-- [Task & effort](#task--effort)
+- [Task &amp; effort](#task--effort)
+<!-- /toc -->
 
 ## Introduction
 
-This proposal introduces an improvement for the installation of applications in user-clusters. It aims to replace the current [Custom Addons](https://docs.kubermatic.com/kubermatic/v2.18/guides/addons/), which are applications a user can install on their own. It will not change the installation process for Default Addons, which are provided by Kubermatic. In order to have a clear distinction between the two, this proposal uses the following terminology hereafter:
+This proposal introduces an improvement for the installation of applications in user clusters. It aims to replace the current [Custom Addons](https://docs.kubermatic.com/kubermatic/v2.18/guides/addons/), which are applications a user can install on their own. It will not change the installation process for Default Addons, which are provided by Kubermatic. In order to have a clear distinction between the two, this proposal uses the following terminology hereafter:
 
 - Application *(new)* -> Optional component that can be installed by the user in their user cluster
 - Default Addons -> Mandatory component that is provided by Kubermatic using the existing Addon mechanism
```

Comment:
I think in this case and hereafter we can use the wording we use the wording [from the documentation](https://docs.kubermatic.com/kubermatic/master/tutorials-howtos/applications/#comparison-to-kkp-addons)